lib/pure/md5.nim

```diff
@@ -97,10 +98,35 @@ proc decode(dest: var openArray[uint8], src: openArray[uint32]) =
     dest[i+3] = uint8(src[j] shr 24 and 0xff'u32)
     inc(i, 4)
 
-proc transform(buffer: pointer, state: var MD5State) =
+template slice(s: string, a, b): openArray[uint8] =
+  when nimvm:
+    var s2 = newSeq[uint8](s.len)
+    for i in 0 ..< s2.len:
+      s2[i] = uint8(s[i])
+    s2
```

Comment:
`toOpenArray` is unimplemented for VM (#15952) and `toOpenArrayByte` for cstring is unimplemented in JS, so here are workarounds